addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -219,6 +228,80 @@ odoo.define('website_slides.quiz', function (require) {
             });
         },
 
+        /**
+         * @private
+         */
+        _onClicksigninJoin: function (event) {
+            event.preventDefault();
+            var self = this;
+            var values = self._getAnswers()
+            if (values.length === self.quiz.questions.length){
+                self._alertHide();
+                if (this.publicUser){
+                    values = {'slide_id': this.slide.id, 'slide_answers':values}
+                    return this._rpc({
+                        route:'/slides/slide/quiz/save_slide_answsers',
+                        params: {
+                            'slide_values': values,
+                        }
+                    }).then(function (){
+                        var url = self._createLoginRedirectUrl()
+                        window.location.href = url;
+                    });
+                }
+            } else {
+                self._alertShow();
+            }
+        },
+        /**
+         * @private
+         * @param
+         */
+        _createLoginRedirectUrl: function(){
+            var urlParts = window.location.pathname.split('/');
+            var url =  urlParts.join('/');
+            var params = {};
+            if (window.location.href.indexOf("fullscreen") > -1) {
+                params.fullscreen=1;
+            }
+            this.redirectURL= _.str.sprintf('%s?%s', url, $.param(params));
+            return _.str.sprintf('/web/login?redirect=%s', encodeURIComponent(this.redirectURL));
+        },
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+        _alertHide: function () {
+            this.$('.o_wslides_js_lesson_quiz_error').addClass('d-none');
+        },
+
+        _alertShow: function (alert_code) {
+            var message = _t('There was an error validating this quiz.');
+            if (! alert_code || alert_code === 'slide_quiz_incomplete') {
+                message = _t('All questions must be answered !');
+            }
+            else if (alert_code === 'slide_quiz_done') {
+                message = _t('This quiz is already done. Retaking it is not possible.');
+            }
+            else if (alert_code === 'public_user') {
+                message = _t('You must be logged to submit the quiz.');
+            }
+            this.$('.o_wslides_js_lesson_quiz_error span:first').html(message);
```

Comment:
You should try to avoid `html()` as much as possible since it introduces XSS attacks points.
`.text()` should work just fine for this case.